### PR TITLE
Issue-1864: Add a rule using the maven-enforcer plugin that requires Maven 3.6.3

### LINKED ISSUE #1864: Add a rule using the maven-enforcer plugin that requires Maven 3.6.3
# Task Description

We should add a Maven Enforcer rule that requires Maven 3.6.3 in order to build the code.

# Tasks

The following tasks will need to be carried out:
* [ ] Add this rule in the [strongbox-parent].
* [ ] Apply this rule to the [strongbox] project.
* [ ] Apply this rule to the [strongbox-web-integration-tests].

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @sbespalov
  * @steve-todorov

[strongbox]: https://github.com/strongbox/strongbox/
[strongbox-parent]: https://github.com/strongbox/strongbox-parent/
[strongbox-web-integration-tests]: https://github.com/strongbox/strongbox-web-integration-tests/

### PR DESCRIPTION
**Pull request by madhukarbharti**
# Pull Request Description

This pull request closes #1864

# Acceptance Test

* [x] Building the code with `mvn clean install -Dintegration.tests` still works.
* [x] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [x] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [x] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [x] No

* Does this pull request require other pull requests to be merged first? 
  * [x] Yes, please see strongbox/strongbox-parent#101
  * [ ] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [x] No

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -213,7 +213,7 @@ pipeline {
                             }
 
                             dir("windows") {
-                                withMavenPlus(maven: 'maven-3.5',
+                                withMavenPlus(maven: 'maven-3.6',
                                               mavenSettingsConfig: '67aaee2b-ca74-4ae1-8eb9-c8f16eb5e534',
                                               mavenOpts: MVN_OPTS,
                                               options: [artifactsPublisher(disabled: true)]) {
```
Comment (steve-todorov) [inline Jenkinsfile:219]:
```suggestion
                                withMavenPlus(mavenSettingsConfig: '67aaee2b-ca74-4ae1-8eb9-c8f16eb5e534',
                                              mavenOpts: MVN_OPTS,
                                              options: [artifactsPublisher(disabled: true)]) {
```

**Comment by steve-todorov** [comment]
[Build passes](https://jenkins.carlspring.org/blue/organizations/jenkins/strongbox%2Fbuilds%2Fstrongbox/detail/PR-1902/14/pipeline/14/ ) - you can now revert the pom versions back.

Thanks for your contribution! :) 

@carlspring this PR needs to be merged since it includes fixes for Travis and Jenkins.

**Review by steve-todorov — APPROVED** [review]
LGTM! Thanks and congratulations on your first merged PRs! :)

**Comment by carlspring** [comment]
Thank you for your contribution! :)

**Pull request merged**